### PR TITLE
Test specs for SmartContractOwnershipRegistry.sol

### PR DESCRIPTION
**Pull request by proxima424**

### CODE REVIEW

File: test/utils/setupHelper.ts
Comment (filmakarov) [inline test/utils/setupHelper.ts:1]: If you want to deploy a SCOwnershipRegistry through setupHelper, you should use hardhat-deployments pattern, like it is done for ECDSA Module in `getEcdsaOwnershipRegistryModule` method. 
It involves creating a deploying script in `/src/deploy/` folder
and then using `.attach` to attach to pre-deployed contract in `setupHelper.ts`

File: test/utils/userOp.ts [hunk 1]
```diff
@@ -402,3 +402,44 @@ export async function makeEcdsaModuleUserOpWithPaymaster(
...
+    },
+    userOpSigner,
+    entryPoint,
+    'nonce'
+  );
+
+  let tempSignature = ethers.utils.defaultAbiCoder.encode(
+    ["bytes","address"],
+    [userOp.signature,ecdsaModuleAddress]
+  )
```
Comment (filmakarov) [inline test/utils/userOp.ts:435]: **temp**Signature name says nothing about the purpose of it. why it is temp? it is not temp, it is just for the SA that is owner of userSA

File: test/utils/userOp.ts [hunk 2]
```diff
@@ -402,3 +402,44 @@ export async function makeEcdsaModuleUserOpWithPaymaster(
   return userOpWithPaymasterData;
 }
 
+
+export async function makeSARegistryModuleUserOp(
```
Comment (filmakarov) [inline test/utils/userOp.ts:406]: I like that you decided to introduce a helper function for such kind of userOps

File: test/module/SmartContractOwnershipRegistry.Module.specs.ts [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+            userSA: userSA,
+            smartAccountOwnerContract1: smartAccountOwnerContract1,
+            smartAccountOwnerContract2: smartAccountOwnerContract2,
+            smartContractOwnershipSetupData: smartContractOwnershipSetupData,
+            randomContract: randomContract
+        }
+    });
+
+    describe("initForSmartAccount(): ", async()=>{
+        it("Calls initForSmartAccount and it successfully initializes userSA with Smart Contract as Owner", async()=>{
```
Comment (filmakarov) [inline test/module/SmartContractOwnershipRegistry.Module.specs.ts:75]: initForSmartAccount(): is already in `describe`, no need to put it in `it`
`describe` contains info about which function and `it` contains info about expected behaviour
so 
initForSmartAccount 
     should successfully initialize userSA with Smart Contract as Owner

File: test/module/SmartContractOwnershipRegistry.Module.specs.ts [hunk 4]
```diff
@@ -0,0 +1,491 @@
...
+                 smartContractOwnershipRegistryModule.address,
+                 ecdsaRegistryModule.address
+            );
+            const tx = await entryPoint.handleOps([userOp], alice.address);
+            await expect(tx).to.emit(entryPoint, "UserOperationRevertReason");
+        });
+    });
+
+    describe("transferOwnership(): ", async()=>{
+        it("Call transferOwnership from userSA and it successfully changes owner", async()=>{
```
Comment (filmakarov) [inline test/module/SmartContractOwnershipRegistry.Module.specs.ts:119]: test name should better describe the test purpose

transferOwnership(): 
         should change owner when called by Smart Account when called from this Smart Account

File: test/module/SmartContractOwnershipRegistry.Module.specs.ts [hunk 5]
```diff
@@ -0,0 +1,491 @@
...
+                 entryPoint,
+                 smartContractOwnershipRegistryModule.address,
+                 ecdsaRegistryModule.address
+            );
+            const tx = await entryPoint.handleOps([userOp],alice.address);
+            await expect(tx).to.not.emit(entryPoint, "UserOperationRevertReason");
+            expect(await smartContractOwnershipRegistryModule.getOwner(userSA.address)).to.be.equal(smartAccountOwnerContract2.address);
+        });
+
+        it("Reverts when trying to set EOA as owner via transferOwnership()", async()=>{
```
Comment (filmakarov) [inline test/module/SmartContractOwnershipRegistry.Module.specs.ts:140]: should revert when trying to set EOA as owner

File: test/module/SmartContractOwnershipRegistry.Module.specs.ts [hunk 6]
```diff
@@ -0,0 +1,491 @@
...
+            );
+
+            const tx = await entryPoint.handleOps([userOp],alice.address);
+            await expect(tx).to.emit(entryPoint, "UserOperationRevertReason");
+            expect(await smartContractOwnershipRegistryModule.getOwner(userSA.address)).to.be.equal(previousOwner);
+        });
+    });
+
+    describe("renounceOwnership(): ", async()=>{
+        it("Should be able to renounce ownership and the new owner should be address(0)", async()=>{
```
Comment (filmakarov) [inline test/module/SmartContractOwnershipRegistry.Module.specs.ts:189]: this is good name

File: test/module/SmartContractOwnershipRegistry.Module.specs.ts [hunk 7]
```diff
@@ -0,0 +1,491 @@
...
+            const userOpHash = getUserOpHash(sendTokenUserOp,entryPoint.address,chainId);
+            await expect(smartContractOwnershipRegistryModule.validateUserOp(sendTokenUserOp,userOpHash)).to.be.revertedWith("NoOwnerRegisteredForSmartAccount");
+
+            const unregisteredSABalanceBefore = await mockToken.balanceOf(unregisteredSA.address);
+            await expect(entryPoint.handleOps([sendTokenUserOp],alice.address)).to.be.revertedWith("FailedOp");
+            expect(await mockToken.balanceOf(bob.address)).to.equal(bobBalanceBefore);
+            expect(await mockToken.balanceOf(unregisteredSA.address)).to.equal(unregisteredSABalanceBefore);
+        });
+
+        it("Reverts when length of user.signature is less than 65 ", async()=>{
```
Comment (filmakarov) [inline test/module/SmartContractOwnershipRegistry.Module.specs.ts:373]: this case is not needed. there's no functionality that checks signature length in SCOwnershipRegistry, so nothing to test here

if in the Smart Account that is owner of userSA there will be not an ECDSA module as validator, this test won't pass.

File: test/module/SmartContractOwnershipRegistry.Module.specs.ts [hunk 8]
```diff
@@ -0,0 +1,491 @@
...
+            const moduleSignature = ethers.utils.defaultAbiCoder.encode(
+                ["bytes","address"],
+                [tempSignature,ecdsaRegistryModule.address]
+            );
+
+            // set msg.sender to be unregisteredSmartAccount instead of userSA.address
+            await expect(smartContractOwnershipRegistryModule.isValidSignatureForAddress(dataHash,moduleSignature,unregisteredSmartAccount)).to.be.revertedWith("NoOwnerRegisteredForSmartAccount");
+        });
+
+        it("Reverts when signature length is less than 65", async()=>{
```
Comment (filmakarov) [inline test/module/SmartContractOwnershipRegistry.Module.specs.ts:450]: same here

**Review by filmakarov — CHANGES_REQUESTED** [review]
Generally good. please address the comments below

File: test/module/SmartContractOwnershipRegistry.Module.specs.ts (hunk 7 above)
Comment (proxima424) [reply test/module/SmartContractOwnershipRegistry.Module.specs.ts:373]: Understood

File: test/utils/userOp.ts (hunk 2 above)
Comment (proxima424) [reply test/utils/userOp.ts:406]: :)

File: test/utils/userOp.ts (hunk 1 above)
Comment (proxima424) [reply test/utils/userOp.ts]: changed name to signatureForECDSAOwnerhsipRegistry

File: test/utils/setupHelper.ts
Comment (proxima424) [reply test/utils/setupHelper.ts:1]: Doneee

**Pull request merged**